homeassistant/components/eq3btsmart/binary_sensor.py

```diff
@@ -71,12 +70,11 @@ def __init__(
         entity_description: Eq3BinarySensorEntityDescription,
     ) -> None:
         """Initialize the entity."""
-
         super().__init__(entry, entity_description.key)
         self.entity_description = entity_description
 
-    @property
-    def is_on(self) -> bool:
-        """Return the state of the binary sensor."""
-
-        return self.entity_description.value_func(self._thermostat.status)
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_is_on = self.entity_description.value_func(self.coordinator.data)
+        super()._handle_coordinator_update()
```

Comment:
To continue my previous comment, just keeping it like the is_on property would eliminate the need for calling the listeners